### DISCUSSION TITLE
Searching for HPX-Application to generate work on multiple Nodes

### DISCUSSION THREAD

**Discussion by leonfse** [STEllAR-GROUP/hpx#5146] [Q&A]
Hello, I am looking for an open source HPX-Application, which I can use to generate work on multiple Nodes, in order to analyse the performance of said application.

Any ideas, suggestions and experiences are very welcome,
Leon

**Comment by AmosLewis** [comment]
Any hpx + cuda + multiple nodes example?

**Comment by hkaiser** [comment]
> Any hpx + cuda + multiple nodes example?

HPX distributed work and CUDA are orthogonal and unrelated. You should be able to combine these two easily.

**Comment by leonfse** [comment]
For me HPX without any other Libraries would be most fitting.
Can any of the Examples in this GitHub Repository put multiple Nodes to work?

**Comment by msimberg** [comment]
@leonfse I suppose the 1D stencil examples in `examples/1d_stencil` would be one (simple) option. I think starting from `1d_stencil_5` they are distributed. https://github.com/STEllAR-GROUP/tutorials/tree/master/examples/03_stencil also contains a 2D stencil example, although it may need some tweaking to make it work again with the latest version of HPX.

**Reply by leonfse** [reply]
@msimberg Thank you very much for the hint! 1d_stencil_5 did not work on two haswell nodes for me, but 1d_stencil_8 did.

**Reply by hkaiser** [reply]
> @msimberg Thank you very much for the hint! 1d_stencil_5 did not work on two haswell nodes for me, but 1d_stencil_8 did.

Yes, 1d_stencil_8 is what you want to play with. I'll have a look at 1d_stencil_5 to see what's wrong.
